spark/src/test/java/com/netflix/iceberg/spark/data/TestParquetAvroReader.java

```diff
@@ -184,7 +184,8 @@ public void testWithOldReadPath() throws IOException {
 
   @Test
   public void testCorrectness() throws IOException {
-    Iterable<Record> records = RandomData.generate(COMPLEX_SCHEMA, 250_000, 34139);
+    // TODO (yifeih): Change the seed back to 34139 after merging https://github.com/apache/parquet-mr/pull/620
+    Iterable<Record> records = RandomData.generate(COMPLEX_SCHEMA, 250_000, 34138);
```

Comment:
@rdblue for SA - we hit another Parquet bug, I think, which is also fixed by https://github.com/apache/parquet-mr/pull/620. We don't think the actual upgrades we're doing in this patch are related to the bug. It's very specific - an array is written, the last value in the array is null, the page is written (without the cached null) and then an [assertion in Parquet](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterBase.java#L87) fails with the pageRowCount being 0 but the repetitionLevel being 1 when nulls are flushed from the previous page.

We don't understand the root cause here, but since the problem goes away from applying the above patch that flushes nulls eagerly, I think we're fine here.